draco/core/task.py

```diff
@@ -381,7 +401,13 @@ def _save_output(self, output):
             tag = output.attrs["tag"] if "tag" in output.attrs else self._count
 
             # Construct the filename
-            outfile = self.output_root + str(tag) + ".h5"
+            name_parts = {
+                "tag": tag,
+                "count": self._count,
+                "task": self.__class__.__name__,
+                "key": self._out_keys[0],
+            }
+            outfile = self.output_name.format(**name_parts)
```

Comment:
`output_name` currently defaults to `None` which I think would raise an error here if left unset. Should it instead default to `""` as `output_root` did?